app/uk/gov/hmrc/tai/connectors/TaxAccountConnector.scala

```diff
@@ -76,14 +76,14 @@ class TaxAccountConnector @Inject()(httpHandler: HttpHandler, servicesConfig: Se
         TaiTaxAccountFailureResponse(e.getMessage)
     }
 
-  def taxCodeIncomes(nino: Nino, year: TaxYear)(implicit hc: HeaderCarrier): Future[TaiResponse] =
+  def taxCodeIncomes(nino: Nino, year: TaxYear)(
+    implicit hc: HeaderCarrier): Future[Either[String, Seq[TaxCodeIncome]]] =
     httpHandler.getFromApiV2(taxAccountUrl(nino.nino, year)) map (
-      json => TaiSuccessResponseWithPayload((json \ "data").as[Seq[TaxCodeIncome]](Reads.seq(taxCodeIncomeSourceReads)))
+      json => Right((json \ "data").as[Seq[TaxCodeIncome]](Reads.seq(taxCodeIncomeSourceReads)))
     ) recover {
-      case e: UnauthorizedException => TaiUnauthorisedResponse(e.getMessage)
```

Comment:
we hve lost the ability to track what exception we have here but from what I can see we never use this and all we do is say
`case e: Exception => ???`
which I have just replaced with
`case Left() => ???`